test/razor/razorIntegrationTests/reference.integration.test.ts

```diff
@@ -0,0 +1,160 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as path from 'path';
+import * as vscode from 'vscode';
+import { beforeAll, afterAll, test, expect, beforeEach } from '@jest/globals';
+import testAssetWorkspace from './testAssets/testAssetWorkspace';
+import * as integrationHelpers from '../../lsptoolshost/integrationTests/integrationHelpers';
+
+integrationHelpers.describeIfDevKit(`Razor References ${testAssetWorkspace.description}`, function () {
+    beforeAll(async function () {
+        if (!integrationHelpers.isRazorWorkspace(vscode.workspace)) {
+            return;
+        }
+
+        await integrationHelpers.activateCSharpExtension();
+    });
+
+    beforeEach(async function () {
+        await integrationHelpers.openFileInWorkspaceAsync(path.join('Pages', 'Definition.razor'));
+    });
+
+    afterAll(async () => {
+        await testAssetWorkspace.cleanupWorkspace();
+    });
+
+    test('Go To Definition', async () => {
+        if (!integrationHelpers.isRazorWorkspace(vscode.workspace)) {
+            return;
+        }
+
+        const position = new vscode.Position(6, 41);
+
+        await integrationHelpers.waitForExpectedResult<vscode.Location[]>(
+            async () =>
+                await vscode.commands.executeCommand(
```

Comment:
is this done in waitForExpectedResults in order to retry?  Generally awaiting the command should return the results

Reply:
Yea, it's just for the retry logic. Since this now is actually using project information instead of misc project it can happen before syncing csharp if the test runs too fast.